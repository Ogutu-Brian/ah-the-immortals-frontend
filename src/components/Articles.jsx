import React, { Component } from 'react';
import '../styles/App.css';

const imageStyle = {
  height: ' auto',
  width: '90%',
  display: 'block',
  margin: '0 30px 15px',
};

class Articles extends Component {
  render() {
    return (
      <div className="flex-container">
        <div className="tags-box">
          <p>Recent</p>
        </div>
        <div className="article">
          <button type="button" className="btn btn-light" data-toggle="modal" data-target=".bd-example-modal-xl">
            Start a post
          </button>
          <div className="card border-primary mb-3">
            <div className="card-body">
              <div className="card-title">
                <h1>Andela Asset Tags</h1>
                <h6 className="card-subtitle text-muted">
                  We (Finance team) are currently conducting an asset verification exercise with each department /cohort to ensure that we clearly tag laptops belonging to Andela.
                </h6>
              </div>
            </div>
            <div className="card-body user-details">
              <img src="https://img.icons8.com/ios/50/000000/user-male-circle-filled.png" alt="avatar" className="  rounded-circle img-responsive " />
              <div className="details">
                <p className="user-details-name">Mary</p>
              </div>
            </div>
            <div className="card-body">
              <h6 className="card-subtitle text-muted">
                We (Finance team) are currently conducting an asset verification exercise....
              </h6>
            </div>
          </div>

          <div
            className="modal fade bd-example-modal-xl" tabIndex="-1" role="dialog"
            aria-labelledby="myExtraLargeModalLabel" aria-hidden="true">
            <div className="modal-dialog modal-xl">
              <div className="modal-content">
                <h1>testing dude</h1>
              </div>
            </div>
          </div>
        </div>
        <div className="tags-box" />
      </div>
    );
  }
}

export default Articles;
