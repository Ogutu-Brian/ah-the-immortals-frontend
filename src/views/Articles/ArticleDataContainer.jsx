import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getAllArticles } from '../../redux/actions/allArticles.action';
import TopRatedArticles from  '../../components/home/TopRatedArticles';
import { sortTopRated } from '../../services/sortArticles';

class Articles extends Component {

  componentDidMount() {
    const { getAllArticles } = this.props;
    getAllArticles();
  }

  render() {
    const { allArticles: { articles } } = this.props;
    const articleRatings = articles.map(article => article);
    if(articleRatings.length !== 0) sortTopRated(articleRatings);
    const topArticles = articleRatings.slice(0, 4);
    return (
      <div>
        <TopRatedArticles
          topArticles={topArticles}
        />
      </div>
    );
  }
}

Articles.propTypes = {
  getAllArticles: PropTypes.func.isRequired,
  allArticles: PropTypes.object.isRequired,
};

const mapStateToProps = ({ allArticles }) => {
  return {
    allArticles: allArticles
  };
};

export default connect(mapStateToProps, { getAllArticles })(Articles);
