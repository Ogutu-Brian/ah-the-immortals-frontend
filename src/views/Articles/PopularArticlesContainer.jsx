import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getAllArticles } from '../../redux/actions/allArticles.action';
import PopularArticles from '../../components/home/PopularArticlesComponent';
import { sortPopular } from '../../services/sortArticles';

class PopularArticlesContainer extends Component {

  componentDidMount() {
    const { getAllArticles } = this.props;
    getAllArticles();
  }

  render() {
    const { allArticles: { articles } } = this.props;
    if(articles.length !== 0) sortPopular(articles);
    const likesCount = articles.slice(0, 4);
    return (
      <div>
        <PopularArticles likesCount={likesCount} />
      </div>

    );
  }
}

PopularArticlesContainer.propTypes = {
  getAllArticles: PropTypes.func.isRequired,
  allArticles: PropTypes.object.isRequired
};

const mapStateToProps = ({ allArticles }) => {
  return {
    allArticles: allArticles
  };
};

export default connect(mapStateToProps, { getAllArticles })(PopularArticlesContainer);
