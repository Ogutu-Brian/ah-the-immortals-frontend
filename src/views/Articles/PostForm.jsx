import React, { Component } from 'react';
import ReactQuill, { Quill } from 'react-quill';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import 'react-quill/dist/quill.snow.css';
import { connect } from 'react-redux';
import Spinner from '../Article/Spinner';
import { addPost } from '../../redux/actions/postActions';



export class PostForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      body: '',
      description: '',
      isLoading: false,
      errors: {}
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(newProps) {
    if (newProps.errors) {
      this.setState({ errors: newProps.errors });
    }
  }



  onSubmit = async (e) => {
    this.setState({ isLoading: true });
    e.preventDefault();
    const { history } = this.props;
    const article = { ...this.state };
    const { addPost } = this.props;
    await addPost(article, (slug) => history.push(`/post/${slug}`), () => this.setState({ isLoading: false }));
  };

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleBodyChange = event => {
    this.setState({
      body: event
    });
  };


  render() {
    const { errors, title, isLoading, description, body } = this.state;


    let Image = Quill.import('formats/image');
    Image.className = 'img-fluid';
    Quill.register(Image, true);

    return (
      <div>
        {
          isLoading ? <Spinner /> :
            (
              <div>
                <br />
                <br />
                <br />
                <div className="row">
                  <div className="col-md-8 offset-md-2">
                    <div className="card card-info">
                      <div className="card-header" />
                      <div className="card-body">

                        <form onSubmit={this.onSubmit}>
                          <div className="form-group">
                            <label htmlFor="Title-name" className="col-form-label">Title</label>
                            <input
                              type="text"
                              className={classnames('form-control', {
                                'is-invalid': errors.title
                              })}
                              name="title" value={title}
                              onChange={this.onChange}
                              id="Title-name"
                              placeholder="Title Of the Article"
                              required
                              error={errors.title}
                            />
                            {errors.title && <div className="invalid-feedback">{errors.title}</div>}
                          </div>
                          <div className="form-group">
                            <label htmlFor="description-name" className="col-form-label">Description</label>
                            <input
                              type="text"
                              className={classnames('form-control', {
                                'is-invalid': errors.description
                              })}
                              name="description"
                              value={description}
                              onChange={this.onChange}
                              id="description-name"
                              placeholder="This is a short summary of the Article"
                              required
                            />
                            {errors.description && <div className="invalid-feedback">{errors.description}</div>}
                          </div>
                          <div className="form-group">
                            <label htmlFor="Body-text" className="col-form-label">Body</label>
                            <ReactQuill
                              modules={PostForm.modules}
                              formats={PostForm.formats}
                              value={body}
                              id="Body-text"
                              placeholder="What do you want to talk about"
                              name="body"
                              onChange={this.handleBodyChange}
                              required
                            />

                          </div>
                          <div className="form-group">
                            <label htmlFor="Tags-name" className="col-form-label">+ Add Tags</label>
                            <input
                              type="text"
                              placeholder="Help the right people see your post "
                              className="form-control"
                              id="Tags-name"
                            />
                          </div>
                          <div className="modal-footer">
                            <button name="post-form" type="submit" className="btn btn-primary">
                            Publish
                            </button>
                            <button type="button" className="btn btn-secondary">cancel</button>
                          </div>
                        </form>

                      </div>
                    </div>
                  </div>
                </div>
                <br />
                <br />
                <br />
              </div>
            )
        }
      </div>


    );
  }
}


PostForm.modules = {
  toolbar: {
    container: [
      [{ font: [] }],
      [{ size: ['small', false, 'large', 'huge'] }],
      ['bold', 'italic', 'underline', 'strike'],
      [{ script: 'sub' }, { script: 'super' }],
      ['blockquote', 'code-block'],
      [{ list: 'ordered' }, { list: 'bullet' }],
      [{ indent: '-1' }, { indent: '+1' }],
      [{ color: [] }, { background: [] }],
      ['link', 'image', 'video'],
      [{ align: [] }],
      ['clean']
    ]
  }
};
PostForm.formats = [
  'header',
  'font',
  'size',
  'italic',
  'underline',
  'strike',
  'bold',
  'blockquote',
  'list',
  'link',
  'image',
  'video',
  'code-block',
  'indent',
  'align',
  'color',
  'background',
  'script',
  'script'
];
PostForm.propTypes = {
  addPost: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  post: state.post.posts,
  errors: state.errors
});

export default connect(mapStateToProps, { addPost })(PostForm);
