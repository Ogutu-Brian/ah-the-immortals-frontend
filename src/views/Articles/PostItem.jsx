import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import Moment from 'react-moment';
import '../../styles/App.css';


class PostItem extends Component {


  render() {
    const { post } = this.props;
    const {created_at} = post;
    return (
      <div className="quotes">
        <div className="box box1">

          <div className="">
            <div className="title-header">
              {post.title}
              <div className="timer">
                <div className="bookmark">
                  <i className="far fa-bookmark" />
                </div>
                <span className="readingTime">
                  {post.readtime}
                </span>
              </div>
            </div>
            <div className="author">
              {post.author.username}
            </div>
            <div className="description">
              {post.description}
              <Link to={`/post/${post.slug}`}>
                .... Read More
              </Link>
            </div>
            <div className="date-read-time">
              <time>
                <Moment format="DD-MMM-YYYY">
                  {created_at}
                </Moment>
              </time>
            </div>
          </div>
        </div>
      </div>
      
     
    );
  }
}



PostItem.propTypes = {

  post: PropTypes.object.isRequired

};

const mapStateToProps = state => ({

});

export default connect(mapStateToProps)(
  PostItem
);
