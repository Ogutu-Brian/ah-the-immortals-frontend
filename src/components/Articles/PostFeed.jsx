import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PostItem from '../../views/Articles/PostItem';

class PostFeed extends Component {
  render() {
    const { posts } = this.props;

    return posts.map(post => <PostItem key={post.slug} post={post} />);
  }
}

PostFeed.propTypes = {
  posts: PropTypes.array.isRequired
};

export default PostFeed;
