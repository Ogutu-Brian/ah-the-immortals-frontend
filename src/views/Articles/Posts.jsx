import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Spinner from '../Article/Spinner';
import PostFeed from '../../components/Articles/PostFeed';
import { getPosts } from '../../redux/actions/postActions';
import '../../styles/App.css';



class Posts extends Component {
  componentDidMount() {
    const { getPosts } = this.props;
    getPosts();
  }

  render() {
    const { post: { posts, loading } } = this.props;
    return (
      <div>
        <div className="post-feed">
          <br />
          {
            (!posts || loading)? <Spinner />: <PostFeed posts={posts} />
          }
        </div>
        <div className="row">
          <div className="col" />
          <div className="col">
            <div>
              <ul className="pagination pagination-lg">
                <li className="page-item disabled">
                  <a className="page-link" href="#">&laquo;</a>
                </li>
                <li className="page-item active">
                  <a className="page-link" href="#">1</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">2</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">3</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">4</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">5</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">&raquo;</a>
                </li>
              </ul>
            </div>

          </div>
          <div className="col" />
        </div>
        <br />
        <br />
        <br />
        <br />
      </div>
     
    );
  }
}

Posts.propTypes = {
  getPosts: PropTypes.func.isRequired,
  post: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  post: state.post,
  errors: state.errors
});

export default connect(mapStateToProps, { getPosts })(Posts);
