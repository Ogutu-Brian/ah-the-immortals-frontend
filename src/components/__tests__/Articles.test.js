import {mount} from 'enzyme';
import React from 'react';
import Articles from '../Articles';


describe('pass articles', () => {
  const wrapper = mount(
    <Articles />
  );
  it('test 1', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
