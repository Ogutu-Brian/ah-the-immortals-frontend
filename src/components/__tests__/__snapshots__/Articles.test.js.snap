// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pass articles test 1 1`] = `ReactWrapper {}`;
